### PR TITLE
Add Dockerfile and workflow

### LINKED ISSUE #933: Providing Docker packages and a Dockerfile for numba-dpex
Providing a Dockerfile that creates an environment with all dependencies will help collaborators and users of `numba-dpex` to quickly setup a working environment to build and test `numba-dpex`. 

The soda-inria group has a Dockerfile (https://github.com/soda-inria/sklearn-numba-dpex/blob/main/docker/Dockerfile) that sets up all needed dependencies from all dpex packages: `dpnp`, `dpctl` and `numba-dpex` and then builds these three packages from source. We can use that Dockerfile as a basis to create a Dockerfile for `numba-dpex`.

The Dockerfile should achieve the following: 

- [ ] Install all needed driver and runtime dependencies
- [ ] Build dpnp and dpctl from source and install them
- [ ] Install the needed llvm-spirv binary from dpcpp (like done in soda-inria/sklearn-numba-dpex)

Once that is done, we should be able to clone numba-dpex and build and run it in the Docker container.

### PR DESCRIPTION
**Pull request by ZzEeKkAa**
Add Dockerfile both for runtime and building with and without gpu drivers

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings? (n/a)
- [x] If this PR is a work in progress, are you filing the PR as a draft?

Workflow will produce these docker tags:
https://github.com/ZzEeKkAa?tab=packages&repo_name=numba-dpex

Action result can be found here:
https://github.com/ZzEeKkAa/numba-dpex/actions/runs/4473683752

There where few issues that prevented docker from building:
- dpctl was not compiling on Debian based machine (fixed in dpctl repo)
- latest graphics drivers fail numba_dpex test for kernel that tries to print from GPU

Fixes #933

### CODE REVIEW

File: docs/user_guides/getting_started.rst [hunk 1]
```diff
@@ -103,6 +103,54 @@ To run the examples:
 
     python numba_dpex/examples/sum.py
 
+Docker
+------
+
```
Comment (diptorupd) [inline docs/user_guides/getting_started.rst:108]: Need to add instruction/information about the other two packages: `builder`, `stages`.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+# Driver args
+# print on gpu is broken for 22.43.24595.30 + igc-1.0.12812.26
+ARG CR_TAG=22.43.24595.30
+ARG IGC_TAG=igc-1.0.12504.5
+ARG CM_TAG=latest
+ARG L0_TAG=latest
+
+# ONEAPI
+ARG ONEAPI_INSTALLER_URL=https://registrationcenter-download.intel.com/akdlm/irc_nas/19079
```
Comment (diptorupd) [inline Dockerfile:13]: Is this something we have keep changing with every oneAPI release?

**Comment by diptorupd** [comment]
Great work @ZzEeKkAa! 

I lightly tested the runtime package and so far so good :+1: We have to see why the two windows CI run failed, they seem to be unrelated to this PR.

File: Dockerfile (hunk 2 above)
Comment (ZzEeKkAa) [reply Dockerfile:13]: I guess so. I'd like to find way how to deal with it, just setting ONEAPI_VERSION, not build number as well

File: docs/user_guides/getting_started.rst (hunk 1 above)
Comment (ZzEeKkAa) [reply docs/user_guides/getting_started.rst:108]: Done.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**